frontend/src/components/tabs/MetricsTab.tsx

```diff
@@ -0,0 +1,194 @@
+ * Some system metrics are: Confusion Matrix, ROC Curve, Scalar, etc.
+ * Detail can be found in https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/dsl/io_types.py
+ * Note that these metrics are only available on KFP v2 mode.
+ */
+export function MetricsTab({ execution }: MetricsTabProps) {
+  // Retriving a list of artifacts associated with this execution,
+  // so we can find the artifact for system metrics from there.
+  const { isLoading: isLoadingArtifacts, error: errorArtifacts, data: artifacts } = useQuery<
+    Artifact[],
+    Error
+  >(['execution_output_artifact', { id: execution.getId() }], () =>
+    getOutputArtifactsInExecution(execution),
+  );
+
+  // artifactTypes allows us to map from artifactIds to artifactTypeNames,
+  // so we can identify metrics artifact provided by system.
+  const {
+    isLoading: isLoadingArtifactTypes,
+    error: errorArtifactTypes,
+    data: artifactTypes,
+  } = useQuery<ArtifactType[], Error>(['artifact_types'], () => getArtifactTypes());
+
+  // This react element produces banner message if query to MLMD is pending or has error.
+  // Once query is completed, it shows actual content of metrics visualization in MetricsSwitcher.
+  const isLoading = isLoadingArtifactTypes && isLoadingArtifacts;
+  return (
+    <div className={commonCss.page}>
+      <div className={padding(20)}>
+        <div>
+          This step corresponds to execution{' '}
+          <Link
+            className={commonCss.link}
+            to={RoutePageFactory.executionDetails(execution.getId())}
+          >
+            "{ExecutionHelpers.getName(execution)}".
+          </Link>
+        </div>
+        {isLoading && <Banner message='Metrics is loading.' mode='info' />}
+        {errorArtifacts && (
+          <Banner
+            message='Error in retriving metrics information.'
+            mode='error'
+            additionalInfo={errorArtifacts.message}
+          />
+        )}
+        {errorArtifactTypes && (
+          <Banner
+            message='Error in retriving artifact types information.'
+            mode='error'
+            additionalInfo={errorArtifactTypes.message}
+          />
+        )}
+        {!isLoading && !errorArtifacts && !errorArtifactTypes && artifacts && artifactTypes && (
+          <MetricsSwitcher artifacts={artifacts} artifactTypes={artifactTypes}></MetricsSwitcher>
+        )}
+      </div>
+    </div>
+  );
+}
+
+/**
+ * Visualize system metrics based on artifact input.
+ */
+function MetricsSwitcher({ artifacts, artifactTypes }: MetricsSwitcherProps) {
+  // system.ClassificationMetrics contains confusionMatrix or confidenceMetrics.
+  // TODO: Visualize confusionMatrix using system.ClassificationMetrics artifacts.
+  // https://github.com/kubeflow/pipelines/issues/5668
+  let classificationMetricsArtifacts = filterArtifactsByType(
+    'system.ClassificationMetrics',
+    artifactTypes,
+    artifacts,
+  );
```

Comment:
We will support more visualization types, I think it's better to loop over the artifacts and figure out how to visualize each one in the original order.

Reply:
That is true, I will follow up on this in the coming PRs. In the current PR's scope, I will keep the current implementation, knowing that it will be rewritten soon.